hack/local-up-cluster.sh

```diff
@@ -348,24 +347,24 @@ cleanup()
   # fi
 
   # Check if the API server is still running
-  [[ -n "${APISERVER_PID-}" ]] && APISERVER_PIDS=$(pgrep -P ${APISERVER_PID} ; ps -o pid= -p ${APISERVER_PID})
-  [[ -n "${APISERVER_PIDS-}" ]] && sudo kill ${APISERVER_PIDS} 2>/dev/null
+  [[ -n "${APISERVER_PID-}" ]] && mapfile -t APISERVER_PIDS < <(pgrep -P "${APISERVER_PID}" ; ps -o pid= -p "${APISERVER_PID}")
+  [[ -n "${APISERVER_PIDS-}" ]] && sudo kill "${APISERVER_PIDS[@]}" 2>/dev/null
```

Comment:
In ./hack/local-up-cluster.sh line 351:
  [[ -n "${APISERVER_PID-}" ]] && APISERVER_PIDS=$(pgrep -P ${APISERVER_PID} ; ps -o pid= -p ${APISERVER_PID})
                                                            ^--------------^ SC2086: Double quote to prevent globbing and word splitting.
                                                                                             ^--------------^ SC2086: Double quote to prevent globbing and word splitting.

In ./hack/local-up-cluster.sh line 352:
  [[ -n "${APISERVER_PIDS-}" ]] && sudo kill ${APISERVER_PIDS} 2>/dev/null
                                             ^---------------^ SC2086: Double quote to prevent globbing and word splitting.

Reply:
Can you share more about the use of `mapfile` here? It looks like the shellcheck message related more to the double quote issues?

Reply:
APISERVER_PIDS is the command line option and must be split on newlines.
It does not work by just quoting and should be defined as an array.
And shellcheck displayed SC2206 when I used () to define APISERVER_PIDS as an array.
So I used mapfile.

https://github.com/koalaman/shellcheck/wiki/SC2206